doc/source/train/user-guides/configuration-overview.rst

```diff
@@ -0,0 +1,31 @@
+Configuration Overview
```

Comment:
I don't think this page actually adds much value over the [API references](https://docs.ray.io/en/master/train/api/api.html#ray-train-configuration) right now. Maybe we can exclude this page for now and if we see a need for it in the future we can add it with sufficient context/guidance for all of the configurations?

Reply:
I was planning to add more in a follow-up, so I'll just make this page an orphan for now with a todo